src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,173 @@
+            t.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> nextNode = isIndexInTheList(index);
+            Node<T> prevNode = nextNode.prev;
+            Node<T> newNode = new Node<>(prevNode, value, nextNode);
+            nextNode.prev = newNode;
+            if (prevNode == null) {
+                head = newNode;
+            } else {
+                prevNode.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        final int sizeOfInputList = list.toArray().length;
+        checkPositionIndex(size);
+        Node<T> prev;
+        Node<T> next;
+        if (size == size) {
+            next = null;
+            prev = tail;
+        } else {
+            next = isIndexInTheList(size);
+            prev = next.prev;
+        }
+        for (Object elementOfInputList : list.toArray()) {
+            T element = (T) elementOfInputList;
+            Node<T> newNode = new Node<>(prev, element, null);
+            if (prev == null) {
+                head = newNode;
+            } else {
+                prev.next = newNode;
+            }
+            prev = newNode;
+        }
+        if (next == null) {
+            tail = prev;
+        } else {
+            prev.next = next;
+            next.prev = prev;
+        }
+        size += sizeOfInputList;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        return isIndexInTheList(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> node = isIndexInTheList(index);
+        T oldValue = node.value;
+        node.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementIndex(index);
+        return deletingTheNode(isIndexInTheList(index));
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> node = head; node != null; node = node.next) {
+            if (object == null) {
+                if (node.value == null) {
+                    deletingTheNode(node);
+                    return true;
+                }
+            } else {
+                if (object.equals(node.value)) {
+                    deletingTheNode(node);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> isIndexInTheList(int index) {
+        Node<T> node = head;
+        for (int i = 0; i < index; i++) {
```

Comment:
The isIndexInTheList name is expected to return a boolean value. In this case, the name `getNode()` or `getNodeByIndex()` would be better. Rename this method throughout the code